cpp/src/arrow/compute/exec/aggregate.cc

```diff
@@ -75,13 +81,13 @@ Result<std::vector<std::unique_ptr<KernelState>>> InitKernels(
     }
 
     KernelContext kernel_ctx{ctx};
-    ARROW_ASSIGN_OR_RAISE(states[i],
-                          kernels[i]->init(&kernel_ctx, KernelInitArgs{kernels[i],
-                                                                       {
-                                                                           in_types[i],
-                                                                           uint32(),
-                                                                       },
-                                                                       options}));
+    // {in_types[i]..., uint32()}
+    agg_in_types.reserve(in_types[i].size() + 1);
```

Comment:
```suggestion
```
Same as above.  Cryptic comment and dubious reserve.

Reply:
Fixed in a manual commit.